### ISSUE TITLE
FEDERATED spqrl request doen't work in Protege 5.5 in sparql tab

### ISSUE THREAD

**Issue by ARomanovna** [protegeproject/protege#1055]
I would like to create the request below to get something from resource http://dbpedia.org/resource/Weather_forecasting
I tried different examples, but it is still no working, what am I doing wrong?

PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX b: <http://www.menthor.net/myontology#>
PREFIX dbp: <https://dbpedia.org>

SELECT DISTINCT  ?date ?value
WHERE {

SERVICE <https://dbpedia.org/sparql> 
{
?date dbp:date ?value .
}                
}
LIMIT 5

**Comment by LorenzBuehmann** [comment]
did you check the log files - it shows an error which indicates that some classes are missing in the classpath which indicates that the bundle doesn't contain all necessary Java libraries. Long story short, doesn't work until somebody is fixing the RDF/SPARQL bundle

**Comment by LorenzBuehmann** [comment]
that some libraries are missing. Anyways

> Long story short, doesn't work until somebody is fixing the RDF/SPARQL bundle

Until that is being fixed, you won't be able to query DBpedia remotely from within Protege. That is the reality

**Comment by mdebellis** [comment]
Just wanted to add to what Lorenz said: there are a lot of features that unfortunately aren't included in either the SPARQL tab or the Snap SPARQL plugin. Of the two I prefer the Snap SPARQL plugin. The UI is a bit nicer, doing color coded highlighting. Also, Snap SPARQL sees changes made by the reasoner (I don't think the SPARQL tab does) and offers you the option to add new triples to the ontology. You can't do INSERT or DELETE in either implementation of SPARQL but you can use CONSTRUCT  in Snap SPARQL and it will offer you the option to add the new triples to the ontology so it functions the same as INSERT.

**Comment by lolive** [comment]
We also have some issues with the current support of SPARQL in Protege 5. As far as i understand, this is related to Sesame 2.7 being subpar. I am wondering whether rebuilding Protege with Sesame 2.9 would be enough to remove those limitations. Anyone has an idea?

**Comment by fragosog** [comment]
Yes, we tried doing this last year when we were upgrading libraries.  Unfortunately upgrading the sesame version above 2.7.x (through 2.9) doesn't work, some functionalities were broken in those versions and a number of our queries were not terminating (tested at first in the GUI and then with standalone programs).  [After testing rdf4j, we wound up sticking with sesame 2.7.12; rdf4j had various performance issues at the time, but we'll go back to it eventually.]

As mentioned above, the sparql plugin needs additional sesame libraries to cover federation, among other sparql 1.1 functionalities. We added the missing jar files, plus made some modifications to the sparql plugin in our fork & it's working very well. The plugin in our fork, however, won't work in protege 5.5, we've bumped the owlapi version and are not using osgi. But hopefully this info saves people's time testing with the higher versions of sesame.

**Comment by schivmeister** [comment]
There was some discussion about this in the past in the respective plug-in projects, see https://github.com/protegeproject/sparql-query-plugin/issues/18 and https://github.com/protegeproject/snap-sparql-query/issues/26. In short, SPARQL 1.1 Federation _may_ be expected to work in the "normal" SPARQL plug-in, but it is _not_ expected to work in SNAP. If you found a solution or workaround, better to report to the respective plug-in projects.
